src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -101,41 +105,43 @@ internal CodeGenIL(IDictionary<string, IList> input, string filePath, ProtoCore.
 
         private (AssemblyBuilder asmbuilder, TypeBuilder tbuilder) CompileAstToDynamicType(List<AssociativeNode> astList, AssemblyBuilderAccess access)
         {
-            compilePass = CompilePass.MethodLookup;
-            // 0. Gather all loaded function endpoints and cache them.
-            foreach (var ast in astList)
-            {
-                DfsTraverse(ast);
-            }
-            // 1. Create assembly builder (dynamic assembly)
-            var asm = BuilderHelper.CreateAssemblyBuilder("DynamicAssembly", false, access);
-            // 2. Create module builder
-            var mod = BuilderHelper.CreateDLLModuleBuilder(asm, "DynamicAssembly");
-            // 3. Create type builder (name it "ExecuteIL")
-            var type = BuilderHelper.CreateType(mod, "ExecuteIL");
-            // 4. Create method ("Execute"), get ILGenerator 
-            var execMethod = BuilderHelper.CreateMethod(type, "Execute",
-                System.Reflection.MethodAttributes.Static | System.Reflection.MethodAttributes.Private, typeof(void), new[] { typeof(IDictionary<string, IList>),
+            writer = new StreamWriter(logPath);
+            using (Disposable.Create(() => { writer.Close(); writer.Dispose(); }))
```

Comment:
from what I can find online, dispose will also close the stream, so I think you're safe to just use a regular using statement over the writer.